src/states/bootstrapping_peer.rs

```diff
@@ -465,11 +469,11 @@ mod tests {
         let mut sel = mpmc::Select::new();
         machine.register(&mut sel);
 
-        // Blocking step for the first one.
-        let op_index = sel.ready();
+        // Step for the first one.
+        let op_index = unwrap!(sel.try_ready());
```

Comment:
so now switched from a blocking step to a non-blocking step?

Reply:
Yes. The reason is that I had this test fail in the past and it resulted in hang. This would instead panic which is preferable.